src/RightsService.Business/Commands/Role/GetRoleCommand.cs

```diff
@@ -0,0 +1,103 @@
+{
+    /// <inheritdoc/>
+    public class GetRoleCommand : IGetRoleCommand
+    {
+        private readonly ILogger<GetRoleCommand> _logger;
+        private readonly IRoleRepository _roleRepository;
+        private readonly IRightRepository _rightRepository;
+        private readonly IRoleInfoMapper _roleInfomapper;
+        private readonly IUserInfoMapper _userInfoMapper;
+        private readonly IRightMapper _rightMapper;
+        private readonly IRequestClient<IGetUsersDataRequest> _usersDataRequestClient;
+
+        private List<UserInfo> GetUsers(List<Guid> userIds, List<string> errors)
+        {
+            string errorMessage = null;
+
+            List<UserInfo> usersInfo = new();
+
+            try
+            {
+                errorMessage = $"Can not get users info for UserIds {string.Join('\n', userIds)}. Please try again later.";
+
+                var usersDataResponse = _usersDataRequestClient.GetResponse<IOperationResult<IGetUsersDataResponse>>(
+                    IGetUsersDataRequest.CreateObj(userIds)).Result;
+
+                if (usersDataResponse.Message.IsSuccess)
+                {
+                    var usersData = usersDataResponse.Message.Body.UsersData;
+
+                    usersInfo = userIds
+                        .Select(x => _userInfoMapper.Map(usersData.First(ud => ud.Id == x)))
+                        .ToList();
+                }
+                else
+                {
+                    _logger.LogWarning(
+                        $"Can not get users. Reason:{Environment.NewLine}{string.Join('\n', usersDataResponse.Message.Errors)}.");
+                }
+            }
+            catch (Exception exc)
+            {
+                errors.Add(errorMessage);
+
+                _logger.LogError(exc, "Exception on get user information.");
+            }
+
+            return usersInfo;
+        }
+
+        public GetRoleCommand(
+            ILogger<GetRoleCommand> logger,
+            IRoleRepository roleRepository,
+            IRightRepository rightRepository,
+            IRoleInfoMapper roleInfoMapper,
+            IUserInfoMapper userInfoMapper,
+            IRightMapper rightMapper,
+            IRequestClient<IGetUsersDataRequest> usersDataRequestClient)
+        {
+            _logger = logger;
+            _roleRepository = roleRepository;
+            _rightRepository = rightRepository;
+            _roleInfomapper = roleInfoMapper;
+            _userInfoMapper = userInfoMapper;
+            _rightMapper = rightMapper;
+            _usersDataRequestClient = usersDataRequestClient;
+        }
+
+        public RoleResponse Execute(Guid roleId)
+        {
+            RoleResponse result = new();
+
+            var allRights = _rightRepository.GetRightsList();
+
+            var dbRole = _roleRepository.Get(roleId);
+
+            var rights = allRights
+                    .Where(x => dbRole.Rights.Select(x => x.RightId).ToList().Contains(x.Id))
```

Comment:
Why do you use contains here?

Reply:
Because there may be more roles than needed